openml/runs/functions.py

```diff
@@ -98,6 +98,13 @@ def run_model_on_task(
     flow : OpenMLFlow (optional, only if `return_flow` is True).
         Flow generated from the model.
     """
+    if avoid_duplicate_runs and not config.apikey:
+        warnings.warn(
+            "avoid_duplicate_runs is set to True, but no API key is set. "
+            "Please set your API key in the OpenML configuration file "
+            "or pass it directly to the function.",
+            DeprecationWarning,
+        )
```

Comment:
```suggestion
        warnings.warn(
            "avoid_duplicate_runs is set to True, but no API key is set. "
            "Please set your API key in the OpenML configuration file, see"
            "https://openml.github.io/openml-python/main/examples/20_basic/introduction_tutorial.html#authentication"
            " for more information on authentication.",
        )
```

Two changes:
 1. I don't think `DeprecationWarning` is appropriate here because we are not informing about changed behavior/signature of the function, I would favor the `UserWarning` (which is the default warning type, so I left that argument implicit).
 2. There is no way to pass the API key to the function, so that help text is incorrect. I updated the text to point to the documentation.